department-of-veterans-affairs/vets-website
# Add green task form to auth patterns form app

## Are you removing, renaming or moving a folder in this PR?


Did you change site-wide styles, platform utilities or other infrastructure?

## Summary

- Builds out the 'green task' for authenticated patterns mock form. This form is only for staging and will not be launched to production
- Adds introduction, veteran info, mailing address, and edit address pages that are similar to the 1010EZR form
- Task buttons to navigate between the task versions are located at the bottom of the screen
- The beginning of the yellow task config has been started, but is mostly just a starter template that was used to verify that the task buttons worked. The purple task has not been started.
- Lots of boilerplate code copied over from the 1010ezr to establish this first mock form, utils, and alerts
- useLocalStorage hook is used to help mock auth. This is still a WIP but works locally as expected.
- redux store established and an action to set a new form config was created so that each task could start a whole new form flow when needed. We use the url structure to determine what config to load.

I realize that this is a massive PR, mostly because we had to copy a lot of code over from the EZR form to get this first task bootstrapped, so let me know if anything seems out of place or could be cleaned up.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/tmf-auth-exp-design-patterns/issues/51

## Testing done

- Tested locally, this application is a mock form and is not being launched to prod

### Steps to test locally:

- pull in branch, make sure the yarn install is recent
-  run the FE in watch mode with `yarn watch --env entry=mock-form-ae-design-patterns`
- run the mock server with `yarn mock-api --responses src/applications/_mock-form-ae-design-patterns/mocks/server.js`
- open the green task at url `http://localhost:3001/mock-form-ae-design-patterns/task-green` and verify that the form loads

## Screenshots

These are all new pages, so we didnt provide before screenshots

Intro page

![page-1](https://github.com/user-attachments/assets/1e2073f4-d607-4d8e-9fb9-0e16957ded22)

---

Page 2 - Uneditable info

![screencapture-localhost-3001-mock-form-ae-design-patterns-task-green-veteran-information-personal-information-2024-08-12-14_14_16](https://github.com/user-attachments/assets/d34e3ffb-02a4-4b64-a148-7a4456f4604d)

---

Page 3 - Editable info for mailing address

![screencapture-localhost-3001-mock-form-ae-design-patterns-task-green-veteran-information-confirm-mailing-address-2024-08-12-14_14_27](https://github.com/user-attachments/assets/aa2a20f3-d47b-4c55-be5e-8683fc11c520)

---

Page 4 - Editing Mailing Address

![screencapture-localhost-3001-mock-form-ae-design-patterns-task-green-veteran-information-edit-mailing-address-2024-08-12-14_14_46](https://github.com/user-attachments/assets/60a8b7c8-ca71-4be2-bafd-f3bd23446571)

---

Page 5 - Task complete (this is just an idea I had)

![screencapture-localhost-3001-mock-form-ae-design-patterns-task-green-complete-2024-08-12-14_15_00](https://github.com/user-attachments/assets/edced6ba-f483-43c8-b68e-3385f0860d89)


## What areas of the site does it impact?

Doesn't impact production site. On staging it impacts the /mock-form-ae-design-patterns url

## Acceptance criteria

- Pages built from figma designs
- Task buttons load up a different task (just yellow and green set up so far)

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
